firmware/seq.cpp

```diff
@@ -0,0 +1,131 @@
+#include "seq.h"
+
+#define INITIAL_VELOCITY 100
+
+Sequencer::Sequencer(Callbacks callbacks) : callbacks(callbacks) {
```

Comment:
Let's install callbacks in the same way callbacks are added for MIDI handling:
Sequencer.setHandleNoteOn() and Sequencer.setHandleNoteOff()

Reply:
We can change it, but it'll add some more complexity then. We'll need to check for `null` each time before calling them, and we increase the dynamic behaviour of the sequencer.
In this commit I just made the callbacks references instead, meaning we can statically be sure they're never null: https://github.com/datomusic/duo-imxrt/pull/114/commits/268aee14448ce2bb22319eb14dc5da9e0c3d891c

Unless we have a real use-case for dynamically switching the callbacks during runtime, I don't think we should have the dynamic behaviour. But, if you still prefer the setters, I'll change it.

Reply:
there is no need for dynamic switching (yet?) but having an interface consistent with other classes is beneficial.

Reply:
I didn't do anything in regards to this yet, but I think it's something we can leave until just before finishing this up since it's more of a stylistic concern.